templates/template-test.js

```diff
@@ -4,13 +4,22 @@ import { readFileSync } from 'fs';
 
 import fn from '../src/SLUG.vl.js';
 
+function * iter(obj) {
```

Comment:
I'm not familiar with the new node generators. Can you describe what's going on here? Is there a way to accomplish this without using a generator?

Reply:
What is going on there is that I created a generator function to create a custom iterable that recursively calls object.entries to loop through objects and its nested object/array values which will then be used in a for of loop to do whatever we want in each iteration (i.e. converting mark string values into object). Can we accomplish this without the use of generator? After researching and trial and errors, i think i found one. Please review and compare the two.

With generator
```js
/* globals describe it */
import * as assert from 'assert';
import { readFileSync } from 'fs';

import fn from '../src/SLUG.vl.js';

function * iter(obj) {
	for (let [key, value] of Object.entries(obj)) {
			if (Object(value) !== value) yield [obj, key, value];
			else yield * iter(value);
	}
}

const name = 'SLUG';

const spec = JSON.parse(readFileSync(`./specs/${name}.vl.json`, 'utf-8'));

// Changes mark fields that are nested in any objects from a string to an object, which is what vega-lite-api creates
for (let [obj] of iter(spec)) {
	if (typeof obj.mark === "string") {
		obj.mark = { type: obj.mark } 
	}
}

describe(`${name}.vg.js`, () => {
	it(`should match spec`, () => {
		const actual = fn();
		assert.deepStrictEqual(actual, spec);
	});
});

```

Without generator
```js
/* globals describe it */
import * as assert from 'assert';
import { readFileSync } from 'fs';

import fn from '../src/SLUG.vl.js';

function convertMarkValues(obj) {
  for (let key in obj) {
    if (typeof obj[key] === "object") {
        convertMarkValues(obj[key]);
    }
    if (key === "mark" && typeof obj[key] === 'string') {
      obj.mark = { type: obj[key] }
    }
  }
}

const name = 'SLUG';

const spec = JSON.parse(readFileSync(`./specs/${name}.vl.json`, 'utf-8'));

// Changes mark fields that are nested in any objects from a string to an object, which is what vega-lite-api creates
convertMarkValues(spec)


describe(`${name}.vg.js`, () => {
	it(`should match spec`, () => {
		const actual = fn();
		assert.deepStrictEqual(actual, spec);
	});
});
```